trace/trace_test.go

```diff
@@ -579,10 +580,10 @@ func TestSetSpanName(t *testing.T) {
 
 func TestSetSpanNameUnsampledSpan(t *testing.T) {
 	var nilSpanData *SpanData
-	span := startSpan(StartOptions{Sampler: NeverSample()})
-	span.SetName("NoopName")
-
-	if want, got := nilSpanData, span.data; want != got {
+	s := startSpan(StartOptions{Sampler: NeverSample()})
+	s.SetName("NoopName")
+	sp, _ := s.internal.(*span)
```

Comment:
here and in other places with similar change: we should panic rather than ignore cast result I think?
```suggestion
	sp := s.internal.(*span)
```

Reply:
Done for test files.

Do you just want to do this in the test file, or for non-test files as well?

Reply:
We just shouldn't ignore errors. I think you handle that properly in non-test files, i.e. check for `if ok { ... }`.
If not, either add proper handling or let it panic.

Reply:
Got it.  I do handle cast failures properly elsewhere.